code/parachain/frame/fnft/src/lib.rs

```diff
@@ -498,14 +530,24 @@ pub mod pallet {
 	}
 
 	impl<T: Config> FinancialNft<AccountIdOf<T>> for Pallet<T> {
-		/// TODO (vim): Assess the probability of collision in generating accounts
-		///   with collection id type u128 and instance id type u128 this definitely collides
-		///   because the seed is longer than the account ID
 		fn asset_account(
 			collection: &Self::CollectionId,
 			instance: &Self::ItemId,
-		) -> AccountIdOf<T> {
-			T::PalletId::get().into_sub_account_truncating((collection, instance))
+		) -> Result<AccountIdOf<T>, DispatchError> {
+			// If seed is already stored
+			let seed = match FinancialNftAssetAccountSeeds::<T>::try_get(collection, instance) {
```

Comment:
not sure. 

if collection. could you just `seed = hash(collection_od, instance_id)`?

Reply:
If the output of the hash is bigger than `u32`, I'd be concerned by the probability of a partial collision of the full hash being a full collision for the account.
However, it's unlikely. I'll examine a few hashing algorithms that we already have supplied and see if any would work well.
This would be a simpler solution.

Reply:
Gonna try and improve this hash function for variable output and move it into a more general support file for use elsewhere. Seems useful.
ebc768226d872b0921902a95b416f3c7a6132ee1

Reply:
2b4b66d3fe39fff98c6cb34c748a5710f7ac616f